modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -218,7 +218,10 @@ interface DOMUtils {
   };
   is: (elm: Node | Node[], selector: string) => boolean;
   add: (parentElm: RunArguments, name: string | Node, attrs?: Record<string, string | boolean | number>, html?: string | Node, create?: boolean) => HTMLElement;
-  create: (name: string, attrs?: Record<string, string | boolean | number>, html?: string | Node) => HTMLElement;
+  create: {
+    <K extends keyof HTMLElementTagNameMap>(name: K, attrs?: Record<string, string | boolean | number>, html?: string | Node): HTMLElementTagNameMap[K];
```

Comment:
Kinda surprised we didn't already have this. Always good to see more types 👍

Reply:
Just a question — do we even need the second type signature? Unlike events and settings, I can't see a way for 3rd party code to end up introducing different strings that might be valid here. Does anything break if you set a requirement that the elements we're creating are actual HTML elements?

Reply:
Yes, you can create custom elements and those won't be in the `HTMLElementTagNameMap` type.

Reply:
Also worth noting you can do `dom.create('BR')` and that's still valid, but `HTMLElementTagNameMap` only contains lower case variants. So removing the second part would be an API breaking change.